### PR TITLE
feature: adds snippets

### PR DESCRIPTION
**Pull request by krpeacock**

### CODE REVIEW

File: snippets.json [hunk 1]
```diff
@@ -0,0 +1,90 @@
+{
+    "For Loop": {
+        "prefix": [
+            "for"
+        ],
+        "body": [
+            "var ${1:i} = 0;",
```
Comment (crusso) [inline snippets.json:7]: This one looks a bit too specialized, why declare both i and j and increment i by j?
```
for (... in Iter.range(...,...)) {
  ...
};
```
or even just
```
for (... in ...) {
  ...
};
```
might be all you want in this.

(The former won't be well-typed unless Iter is imported...)

File: snippets.json [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    "Type Variant": {
+        "prefix": [
+            "type-variant"
+        ],
+        "body": [
+            "type ${1:Name} = {",
+            "\t#${2:Attribute};",
+            "};"
+        ],
+        "description": "a variable type block"
```
Comment (crusso) [inline snippets.json:44]:
```suggestion
        "description": "a variant type block"
```

**Review by crusso — COMMENTED** [review]
Thanks for starting on this. Some minor comments

File: snippets.json [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    },
+    "Import": {
+        "prefix": [
+            "import"
+        ],
+        "body": [
+            "import ${1:Module} \"mo:base/${1}\";"
+        ]
+    }
+}
```
Comment (o0x) [inline snippets.json:90]:
```suggestion
    "Switch Statement": {
        "prefix": [
            "switch"
        ],
        "body": [
            "switch($1) {",
            "\tcase($2) {$0};",
            "\tcase($3) { };",
            "};"
        ]
    },
}
```

**Review by o0x — CHANGES_REQUESTED** [review]

**Review by ikincihallac — APPROVED** [review]

**Comment by rvanasa** [comment]
Merging in #106. 

@krpeacock, let me know if you have any suggestions from your work on this original PR.

**Pull request closed without merging**